spanner/README.md

```diff
@@ -31,4 +31,87 @@ row, err := client.Single().ReadRow(ctx, "Users",
 if err != nil {
 	log.Fatal(err)
 }
+```
+
+### Session Leak
+A `Client` object of the Client Library has a limit on the number of maximum sessions. For example the
+default value of `MaxOpened`, which is the maximum number of sessions allowed by the session pool in the 
+Golang Client Library is 400. You can configure these values at the time of
+creating a `Client` by passing custom `SessionPoolConfig` as part of `ClientConfig`. When all the sessions are checked
+out of the session pool, every new transaction has to wait until a session is returned to the pool.
+If a session is never returned to the pool (hence causing a session leak), the transactions will have to wait
+indefinitely and your application will be blocked.
+
+#### Common Root Causes
+The most common reason for session leaks in the Golang client library are:
+1. Not stopping a `RowIterator` that is returned by `Query`, `Read` and other methods. Always use `RowIterator.Stop()` to ensure that the `RowIterator` is always closed.
+2. Not closing a `ReadOnlyTransaction` when you no longer need it. Always call `ReadOnlyTransaction.Close()` after use, to ensure that the `ReadOnlyTransaction` is always closed.
+3. Not closing a `BatchReadOnlyTransaction` when you no longer need it. Always call `BatchReadOnlyTransaction.Close()` after use, to ensure that the `BatchReadOnlyTransaction` is always closed.
```

Comment:
For a different PR: We should change the implementation of `BatchReadOnlyTransaction` to not use any sessions from the pool, and instead create a separate session.

Reply:
`BatchReadOnlyTransaction` already creates its own session and do not use any sessions from the pool. Sorry for the confusion.
I have removed this line.